viewing_party/party.py

```diff
@@ -1,23 +1,126 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    # Only create/return a movie dict if all arguments are valid (truthy)
+    # Assumes rating is greater than 0
+    if all([title, genre, rating]):
+        return {
+            "title": title,
+            "genre": genre,
+            "rating": rating,
+        }
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    watched_list = user_data["watched"]
+    watchlist = user_data["watchlist"]
+
+    # Iterate through each movie in watchlist, looking for a match to the argument title
+    for index, movie in enumerate(watchlist):
+        if movie["title"] == title:
+            # Move movie dict from one list to the other
+            transfer_movie = watchlist.pop(index)
+            watched_list.append(transfer_movie)
+            break
+    # Return argument passed in, which may have been updated by for loop
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    ratings = []
+    for movie in user_data["watched"]:
+        # All values should be float, no need to cast them to float
+        ratings.append(movie["rating"])
+    
+    if not ratings:
+        return 0.0
+    else:
+        # Average rating
+        return sum(ratings) / len(ratings)

```

Comment:
Love the efficiency in this bit of code, especially how you return the calculation of average rating on L52 directly!